### PR TITLE
Standardize code style to Black

### PR DESCRIPTION
**Pull request by anishathalye**
[Black](https://black.readthedocs.io/) is the most popular Python code auto-formatting tool, and it's used by many major projects. It supports minimal configuration (similar to tools like `gofmt`), so that this can become a standard code formatting style across Python projects. The only knob tuned in this pull request is the maximum line length, which is changed to 100 from the default 88. (I don't have strong opinions on maximum line length.)

This PR also updates the CI configuration to check code formatting.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[tool.black]
+line-length = 100
```
Comment (jwmueller) [inline pyproject.toml:2]: How do you feel about 120 :)

**Review by jwmueller — APPROVED** [review]
Lgtm!  With this PR, should we also update CONTRIBUTING.md to say Black is being used with link to instructions on how devs can run Black check locally before they open their own PR?

Another consideration is whether we should merge this before @cgnorthcutt 
refactor PR.  @cgnorthcutt will that affect you?

File: pyproject.toml (hunk 1 above)
Comment (anishathalye) [reply pyproject.toml:2]: No strong opinions here. As a test, I tried converting a bunch of my personal repositories to Black, and I tried 88, 100, and 120. In my opinion, 100 looked the most readable.

**Comment by anishathalye** [comment]
> Lgtm! With this PR, should we also update CONTRIBUTING.md to say Black is being used with link to instructions on how devs can run Black check locally before they open their own PR?

Yes; done.

> Another consideration is whether we should merge this before @cgnorthcutt refactor PR. @cgnorthcutt will that affect you?

Yes, let's wait. Merging this PR first will make merging Curtis's work at least slightly more annoying, whereas this PR can easily be rebased on top of Curtis's work and re-running Black. There's no hurry to merge this first.

**Comment by anishathalye** [comment]
Rebased on top of #128, which was just squashed and merged. Right now is the least painful time to apply this kind of global formatting change, while there are no other big PRs pending.

**Comment by jwmueller** [comment]
LGTM go ahead and merge whenever! I agree now is least painful time.

**Review by cgnorthcutt — APPROVED** [review]
LGTM

**Pull request merged**